### PR TITLE
feat: supports critical as a valid severity threshold

### PR DESCRIPTION
**Pull request by cfereday**
As part of work being done APOLLO-97 this ensures critical becomes a valid severity threshold to use in reporting. The CLI test currently returns critical severities from 1.456.0 onwards with sarif support from version 1.514.0.

### CODE REVIEW

File: snykTask/src/__tests__/test-severity-threshold.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    Severity.CRITICAL,
+    Severity.HIGH,
+    Severity.MEDIUM,
+    Severity.LOW
+  ];
+  test("returns true if invalid severity threshold", () => {
+    const isInvalid = isNotValidThreshold("hey");
+    expect(isInvalid).toBe(true);
+  });
+  test("returns false for all valid severity thresholds", () => {
```
Comment (thisislawatts) [inline snykTask/src/__tests__/test-severity-threshold.ts:14]: **suggestion (non-blocking):** Use jest's `each` for easier reading of test output

https://github.com/snyk/snyk-azure-pipelines-task/pull/101/files

File: snykTask/src/index.ts [hunk 2]
```diff
@@ -88,7 +95,7 @@ function parseInputArgs(): TaskArgs {
     taskArgs.severityThreshold = taskArgs.severityThreshold.toLowerCase();
     if (isNotValidThreshold(taskArgs.severityThreshold)) {
       const errorMsg =
-        "If set, severity threshold must be 'high' or 'medium' or 'low' (case insensitive). If not set, the default is 'low'.";
+        "If set, severity threshold must be 'critical' or 'high' or 'medium' or 'low' (case insensitive). If not set, the default is 'low'.";
```
Comment (thisislawatts) [inline snykTask/src/index.ts:98]: **nit-pick (non-blocking):** We have introduced an enum. Let's use it when building the errorMsg string.

**Review by thisislawatts — APPROVED** [review]

File: snykTask/src/index.ts (hunk 2 above)
Comment (cfereday) [reply snykTask/src/index.ts]: Good idea thanks, have incorporated

File: snykTask/src/__tests__/test-severity-threshold.ts (hunk 1 above)
Comment (cfereday) [reply snykTask/src/__tests__/test-severity-threshold.ts]: Very nice idea, thanks  - I've incorporated 😃

File: snykTask/src/__tests__/test-severity-threshold.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    Severity.LOW
+  ];
+  test("returns true if invalid severity threshold", () => {
+    const isInvalid = isNotValidThreshold("hey");
+    expect(isInvalid).toBe(true);
+  });
+
+  describe.each(validSeverityThresholds)(
+    "isNotValidThreshold(%s)",
+    (level, expected) => {
```
Comment (maxjeffos) [inline snykTask/src/__tests__/test-severity-threshold.ts:17]: Something isn't quite right about this
image

Try this:
```
  describe.each(validSeverityThresholds)(
    "isNotValidThreshold(%s)",
    (level) => {
      test(`returns false for '${level}'`, () => {
        expect(isNotValidThreshold(level)).toBe(false);
      });
    }
  );
```

**Review by maxjeffos — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ This extension requires that Node.js and npm be installed on the build agent. Th
 | dockerImageName | The name of the container image to test. | yes, if container image test | none | string |
 | dockerfilePath | The path to the Dockerfile corresponding to the `dockerImageName` | yes, if container image test | none | string |
 | targetFile | Applicable to application type tests ony. The path to the manifest file to be used by Snyk. Should only be provided if non-standard. | no | none | string |
-| severityThreshold | The severity-threshold to use when testing. By default, issues of all severity types will be found. | no | "low" | string: "low" or "medium" or "high" |
+| severityThreshold | The severity-threshold to use when testing. By default, issues of all severity types will be found. | no | "low" | string: "low" or "medium" or "high" or "critical"|
```
[the remark this answers is not in the document]
Comment (cfereday) [reply README.md:28]: Thanks, good spot 🦅 - have amended that .

File: snykTask/src/index.ts [hunk 5]
```diff
@@ -101,13 +107,14 @@ function parseInputArgs(): TaskArgs {
   return taskArgs;
 }
 
-const isNotValidThreshold = (threshold: string) => {
+export const isNotValidThreshold = (threshold: string) => {
```
[the remark this answers is not in the document]
Comment (cfereday) [reply snykTask/src/index.ts:110]: Yes ok that makes sense. Given the fn has moved to `task-lib` I've also consolidated the test file to be in the same `__tests__/task-lib.test.ts` file & deleted the original new file just for testing `isNotValidThreshold`.

File: snykTask/src/__tests__/test-severity-threshold.ts (hunk 3 above)
Comment (cfereday) [reply snykTask/src/__tests__/test-severity-threshold.ts]: Oh right good spot, I've incorporated your suggestion

**Review by maxjeffos — APPROVED** [review]

**Pull request merged**